csi/controller_server.go

```diff
@@ -1287,24 +1296,22 @@ func (cs *ControllerServer) waitForBackupControllerSync(volumeName, snapshotName
 // (and in particular, its name) as quickly as possible and in any state.
 func (cs *ControllerServer) getBackup(volumeName, snapshotName string) (*longhornclient.Backup, error) {
```

Comment:
As we discussed yesterday, this function will return `nil, nil` if the backup is not found. I think if this behavior may be unexpected for some developers then leads to some NPE issues. Do we need to change this or at least add a comment for this behavior?

Reply:
Thank you. I added the comment. 

Looks like we check for return backup to be different than `nil` everywhere we call this fuction so I would prefer to keep the current logic. Ref:
1. https://github.com/longhorn/longhorn-manager/blob/469a446b29588bdaf9104f343c585c8c3c1cf9ee/csi/controller_server.go#L811-L832
2. https://github.com/longhorn/longhorn-manager/blob/469a446b29588bdaf9104f343c585c8c3c1cf9ee/csi/controller_server.go#L1247-L1256
3. https://github.com/longhorn/longhorn-manager/blob/469a446b29588bdaf9104f343c585c8c3c1cf9ee/csi/controller_server.go#L1273-L1280

Reply:
Let's improve this later, need to tackle it eventually to prevent similar issues. 

cc @derekbit